esiclient/v1/node_baremetal.py

```diff
@@ -0,0 +1,84 @@
+#   Licensed under the Apache License, Version 2.0 (the "License"); you may
+#   not use this file except in compliance with the License. You may obtain
+#   a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#   WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#   License for the specific language governing permissions and limitations
+#   under the License.
+
+import logging
+
+import openstack
+from osc_lib.command import command
+from osc_lib import exceptions
+from osc_lib.i18n import _
+
+
+class MDCBaremetalNodeList(command.Lister):
+    """List baremetal nodes from multiple OpenStack instances"""
+
+    log = logging.getLogger(__name__ + ".List")
+
+    def get_parser(self, prog_name):
+        parser = super(MDCBaremetalNodeList, self).get_parser(prog_name)
+        parser.add_argument(
+            '--clouds',
+            dest='clouds',
+            metavar='<clouds>',
+            nargs="+",
+            help=_("Specify the cloud to use from clouds.yaml.")
+        )
+
+        parser.add_argument(
+            '--regions',
+            dest='regions',
+            metavar='<regions>',
+            nargs="+",
+            help=_("Specify the region to use from clouds.yaml. Must specify "
+                   "regions in order corresponding to clouds passed with "
+                   "--clouds flag."
+                   )
+        )
+
+        return parser
+
+    def take_action(self, parsed_args):
+
+        columns = ['Cloud', 'Region', 'UUID', 'Name', 'Instance UUID', 'Power State',
+                   'Provisioning State', 'Maintenance']
+        data = []
+
+
+        if parsed_args.clouds is None and parsed_args.regions is None:
+            cloud_regions = openstack.config.loader.OpenStackConfig().get_all_clouds()
+            
+            for c in cloud_regions:
+
+                #cloud = openstack.connect(config=c)
+                cloud = openstack.connect(cloud=c._name, region=c.config['region_name'])
```

Comment:
Is the name value retrievable from c.config? I'm slightly uncomfortable using an "_" attribute.

Reply:
it is not